### ISSUE TITLE
vmagent unable to handle "ERROR" watch event on Kubernetes

### ISSUE THREAD

**Issue by aayjaychan** [VictoriaMetrics/VictoriaMetrics#1150]
**Describe the bug**

when using Kubernetes service discovery, sometimes Kubernetes API returns WatchEvent with `"type": "ERROR"`.  when this happens, vmagent will flood log output and get stuck retrying.

**Version**

```
$ ./vmagent-prod --version
vmagent-20210317-001105-tags-v1.56.0-0-g20ac89c4e
```

**Additional context**

sampled log:
```
2021-03-24T06:41:21.082Z        error   VictoriaMetrics/lib/promscrape/discovery/kubernetes/api_watcher.go:522  error when reading WatchEvent stream from "https://10.0.0.1:443/api/v1/services?watch=1&allowWatchBookmarks=true&timeoutSeconds=1620&resourceVersion=35785": unexpected WatchEvent type "ERROR" for role "service"
2021-03-24T06:41:21.131Z        error   VictoriaMetrics/lib/promscrape/discovery/kubernetes/api_watcher.go:522  error when reading WatchEvent stream from "https://10.0.0.1:443/api/v1/endpoints?watch=1&allowWatchBookmarks=true&timeoutSeconds=1620&resourceVersion=64611": unexpected WatchEvent type "ERROR" for role "endpoints"
2021-03-24T06:41:21.463Z        error   VictoriaMetrics/lib/promscrape/discovery/kubernetes/api_watcher.go:522  error when reading WatchEvent stream from "https://10.0.0.1:443/api/v1/nodes?watch=1&allowWatchBookmarks=true&timeoutSeconds=1620&resourceVersion=939377": unexpected WatchEvent type "ERROR" for role "node"
2021-03-24T06:41:21.463Z        error   VictoriaMetrics/lib/promscrape/discovery/kubernetes/api_watcher.go:522  error when reading WatchEvent stream from "https://10.0.0.1:443/api/v1/pods?watch=1&allowWatchBookmarks=true&timeoutSeconds=1620&resourceVersion=64650": unexpected WatchEvent type "ERROR" for role "pod"
```
when following one of the urls, this is the response:
```
{"type":"ERROR","object":{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"too old resource version: 2055309 (2508079)","reason":"Expired","code":410}}
```

**Comment by valyala** [comment]
The issue must be fixed in the commit 9761ffd1611b1268bda22061a41cc04cbdf3b6ac . @aayjaychan , could you build `vmagent` from this commit and verify whether it works as expected. Build instructions are the following:
```
git clone https://github.com/VictoriaMetrics/VictoriaMetrics
git checkout 9761ffd1611b1268bda22061a41cc04cbdf3b6ac
make vmagent
```

The built vmagent should be available in `bin` directory.

**Comment by aayjaychan** [comment]
thanks for the fast response.  after running the new version for a few hours, i can confirm that the counter `vm_promscrape_discovery_kubernetes_stale_resource_versions_total` increases while metrics from new pods can still be picked up.  the patch is working.
